src/bytes.js

```diff
@@ -0,0 +1,1458 @@
+
+Sk.builtin.bytes.prototype["$r"] = function () {
+    var ret;
+    var i;
+    var num;
+    ret = "";
+    for (i = 0; i < this.v.byteLength; i++) {
+        num = this.v[i];
+        if ((num < 9) || (num > 10 && num < 13) || (num > 13 && num < 32) || (num > 126)) {
+            ret += makehexform(num);
+        } else if (num === 9 || num === 10 || num === 13 || num === 39 || num === 92) {
+            switch (num) {
+                case 9:
+                    ret += "\\t";
+                    break;
+                case 10:
+                    ret += "\\n";
+                    break;
+                case 13:
+                    ret += "\\r";
+                    break;
+                case 39:
+                    ret += "\\'";
+                    break;
+                case 92:
+                    ret += "\\\\";
+                    break;
+            }
+        } else {
+            ret += String.fromCharCode(num);
+        }
+    }
+    ret = "b'" + ret + "'";
+    return new Sk.builtin.str(ret);
+};
+
+Sk.builtin.bytes.prototype.mp$subscript = function (index) {
+    var ret;
+    var i;
+    if (Sk.misceval.isIndex(index)) {
+        i = Sk.misceval.asIndex(index);
+        if (i !== undefined) {
+            if (i < 0) {
+                i = this.v.byteLength + i;
+            }
+            if (i < 0 || i >= this.v.byteLength) {
+                throw new Sk.builtin.IndexError("index out of range");
+            }
+            return new Sk.builtin.int_(this.v[i]);
+        }
+    } else if (index instanceof Sk.builtin.slice) {
+        ret = [];
+        index.sssiter$(this.v.byteLength, (i) => {
+            ret.push(this.v[i]);
+        });
+        return new Sk.builtin.bytes(ret);
+    }
+
+    throw new Sk.builtin.TypeError("byte indices must be integers, not " + Sk.abstr.typeName(index));
+};
+
+Sk.builtin.bytes.prototype.ob$eq = function (other) {
+    if (this === other) {
+        return Sk.builtin.bool.true$;
+    } else if (!(other instanceof Sk.builtin.bytes)) {
+        return Sk.builtin.NotImplemented.NotImplemented$;
+    } else if (this.v.byteLength != other.v.byteLength) {
+        return Sk.builtin.bool.false$;
+    }
+
+    for (let i = 0; i < this.v.byteLength; i++) {
+        if (this.v[i] != other.v[i]) {
+            return Sk.builtin.bool.false$;
+        }
+    }
+
+    return Sk.builtin.bool.true$;
+};
+
+Sk.builtin.bytes.prototype.ob$ne = function (other) {
+    const ret = this.ob$eq(other);
+    if (ret === Sk.builtin.NotImplemented.NotImplemented$) {
+        return ret;
+    }
+    return Sk.misceval.isTrue(ret) ? Sk.builtin.bool.false$ : Sk.builtin.bool.true$;
+};
+
+Sk.builtin.bytes.prototype.sq$length = function () {
+    return this.v.byteLength;
+};
+
+Sk.builtin.bytes.prototype.bytes_copy_ = function () {
+    var i;
+    var final;
+    final = [];
+    for (i = 0; i < this.v.byteLength; i++) {
+        final.push(this.v[i]);
+    }
+    return new Sk.builtin.bytes(final);
+};
```

Comment:
do we still need this function? I couldn't see that it gets called.

Reply:
It's no longer used.  Removed.